components/Video.tsx

```diff
@@ -0,0 +1,16 @@
+import { HTMLAttributes } from 'react'
+
+export type VideoProps = { src: string } & HTMLAttributes<HTMLVideoElement>
```

Comment:
This seems to be an accident or an overeager GitHub Copilot suggestion.

![image](https://user-images.githubusercontent.com/15332326/183062979-8f71379a-2d99-49fe-a250-b465a3e2b62e.png)

We expose a type promising we'll accept the same props as `<video />` intrinsic element, so
if we pass an `onClick` we'll expect `event.currentTarget` to be `HTMLVideoElement` containing its usual props like `autoplay`.

But in the line 7, we're not spreading the props into a `<video />`, we have a `<figure />` there, so the `event.currentTarget` won't be `HTMLVideoElement`.

We can fix this easily by changing the type like this ⬇️

```suggestion
export type VideoProps = { src: string } & HTMLAttributes<HTMLElement>
```

It also seems non-obvious to me that `<Video />` does not render `<video />` inside.

As this component is only ever used with YouTube video embeds, what do you think about renaming it to **`<Embed />`**?

Reply:
As a side note, if we wanted to spread props to the intrinsic `<video />` element, we could use `React.ComponentPropsWithoutRef<'video'>` or `React.VideoHTMLAttributes<HTMLVideoElement>`.

![image](https://user-images.githubusercontent.com/15332326/183064755-7c77ab9e-b435-4f23-907d-d7364dd406e6.png)

Reply:
How about `<VideoEmbed />`?

Reply:
`<Embed />` could mean something other than videos. Tweets can be embedded, for instance.

Reply:
Yes, `<VideoEmbed />` would work. Also, for me having a video in the naming is self-explanatory that this is for video than anything else.

Reply:
Hey @hasparus , I have updated it to `<HTMLElement>` and thank you so much explaining it so well. I am still learning about types, and typescript really. So this was really helpful to understand why we need to use `<HTMLElement>` and not `<HTMLVideoElement>`

Reply:
What do you both think about `<EmbedVideo />` ? This could also work. Let me know which one we prefer so I will update it to that. Thank you.

Reply:
@alexandriaroberts – Personally, I prefer `VideoEmbed` since it reads better (like in a sentence, you would say "a video embed" or "an embedded video" but not "an embed video") + it starts with the most important thing about it IMO (at least from the content's perspective), which is that it renders a video.

Reply:
> which is that it renders a video.

well... it technically renders a website :P 

```
<Video src="https://twitframe.com/show?url=https%3A%2F%2Ftwitter.com%2Fjack%2Fstatus%2F20" />
```

This would work, right?

Reply:
> This would work, right?

@hasparus – Haha, yes, that would work. We could certainly add some validation for the `src` prop, but we wouldn't want to make it YouTube-only; for instance, we want to have [a Loom embed](https://github.com/graphprotocol/docs/pull/177/files#diff-6c9feb46d631ce72b595a06017b1048c88cb2566b4cfe7475f3aea3680a41f9fR52) on one page, and I could also see us wanting to embed a Vimeo or something else. It's not that component doesn't care about what it renders, but I don't see the point in _not_ having that flexibility; as long as it renders a 16:9 aspect ratio iframe (and we could make that customizable at some point, but it would definitely remain the default) I think we're good.

> I could also see a component like `<YouTubeEmbed video="dQw4w9WgXcQ" />`

That's a good idea! I like that it prevents a potential developer mistake, which is to copy the regular YouTube URL for a video instead of the embed one. If we follow a previously established convention, we could make it a more specific variant of `Video`, e.g. `<Video.YouTube video="dQw4w9WgXcQ" />`.

Reply:
Actually, maybe a `<VideoEmbed youtube="dQw4w9WgXcQ" />`?

Reply:
I like that! Then we can easily add "support" for other services.

Reply:
Okay, I understand what you mean.🙂 I have updated it will `VideoEmbed` and will push the code now.

Reply:
Thanks Lexie! I will add the `youtube` prop suggested by Piotr (as an alternative to `src` – if you set `src` it will override `youtube`). I think it's a great idea.